ddtrace/contrib/celery/utils.py

```diff
@@ -90,7 +92,7 @@ def detach_span(task, task_id, is_publish=False):
         return
 
     # DEV: See note in `attach_span` for key info
-    weak_dict.pop((task_id, is_publish), None)
+    del weak_dict[(task_id, is_publish)]
```

Comment:
I wonder if this should be wrapped in `try ... except KeyError:` 🤔

Reply:
Write a test case? 😉